orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -145,6 +147,8 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
       transformers.emplace_back(std::make_unique<ConstantFolding>(
           execution_provider, false /*skip_dequantize_linear*/, compatible_eps, excluded_initializers));
       transformers.emplace_back(std::make_unique<ReshapeFusion>(compatible_eps));
+      // Put fine-grained optimizers after ReshapeFusion to avoid it breaks the strong patterns it defines.
```

Comment:
Sure. Refined the comment to :

```
      // Put fine-grained optimizer (e.g. ShapeOptimizer) after ReshapeFusion to avoid it breaks the strong patterns
      // it defines. ReshapeFusion depends on subgraph pattern matching and do replacement accordingly, ShapeOptimizer
      // potentially will optimize out some nodes defined in the subgraph patterns. So we put it after ReshapeFusion.
```